x-pack/lib/monitoring/inputs/metrics.rb

```diff
@@ -211,9 +211,18 @@ def fetch_global_stats
     end
 
     def extract_cluster_uuids(stats)
-      result = stats.extract_metrics([:stats, :pipelines, :main, :config], :cluster_uuids)
-      if result && !result[:cluster_uuids].empty?
-        cluster_uuids = result[:cluster_uuids]
+      cluster_uuids = []
+      agent.running_pipelines.each do |pipeline_id, _|
+        unless pipeline_id.to_sym == :".monitoring-logstash"
+          path = [:stats, :pipelines, pipeline_id.to_sym, :config]
+          found_cluster_uuids = stats.extract_metrics(path, :cluster_uuids)
+          if found_cluster_uuids && !found_cluster_uuids[:cluster_uuids].empty?
+            cluster_uuids |= found_cluster_uuids[:cluster_uuids]
+          end
+        end
+      end
```

Comment:
Since we have the pipeline object at hand, we might as well just fetch the cluster uuids directly from it.
This way we can get rid of the :cluster_uuid gauge metric that has been causing the warning we see at startup.

```suggestion
      cluster_uuids = agent.running_pipelines.flat_map do |_, pipeline|
        next if pipeline.system?
        pipeline.resolve_cluster_uuids
      end.compact
```